rotkehlchen/db/accounting_rules.py

```diff
@@ -225,9 +225,31 @@ def missing_accounting_rules(self, events: list[HistoryBaseEntry]) -> list[bool]
         ]
         missing_accounting_rule: list[bool] = []
         with self.db.conn.read_ctx() as cursor:
-            for idx in range(len(events)):
+            for idx, event in enumerate(events):
                 row = cursor.execute(query, bindings[idx]).fetchone()
-                missing_accounting_rule.append(row[0] == 0)
+                if (is_missing_rule := row[0] == 0) is False:
+                    missing_accounting_rule.append(is_missing_rule)
+                    continue
+
+                if (  # check for cases that could use accounting_treatment
+                    event.event_type == HistoryEventType.TRADE and
+                    event.event_subtype in (HistoryEventSubType.RECEIVE, HistoryEventSubType.FEE)
+                ):
+                    cursor.execute(
```

Comment:
This is deep inside a for loop of all events. So this happens for many events. Can't you gather them all in one query and run it once only?

Also though I think I kind of understand what you are doing I would need a few more comments explaining what you mean by "could use accounting_treatment".